cmd/agent/daemon/enrichment/service.go

```diff
@@ -5,117 +5,62 @@ import (
 	"runtime/debug"
 
 	castpb "github.com/castai/kvisor/api/v1/runtime"
-	"github.com/castai/kvisor/cmd/agent/daemon/metrics"
+	"github.com/castai/kvisor/pkg/ebpftracer/events"
 	"github.com/castai/kvisor/pkg/ebpftracer/types"
 	"github.com/castai/kvisor/pkg/logging"
 )
 
 type EnrichRequest struct {
-	Event     *castpb.Event
-	EbpfEvent *types.Event
+	Event *castpb.Event
 }
 
 type EventEnricher interface {
 	// Enrich will add additional data to the provided Event.
-	Enrich(context.Context, *EnrichRequest)
+	Enrich(ctx context.Context, in *types.Event, out *castpb.ContainerEvent)
 
 	// EventsTypes returns a slice of event types, this enricher reacts to.
-	EventTypes() []castpb.EventType
+	EventTypes() []events.ID
 }
 
 type Config struct {
-	WorkerCount    int
 	EventEnrichers []EventEnricher
 }
 
 func NewService(log *logging.Logger, cfg Config) *Service {
 	return &Service{
 		log:            log.WithField("component", "enrichment"),
-		eventsQueue:    make(chan *EnrichRequest, 1000),
-		outQueue:       make(chan *castpb.Event, 1000),
 		cfg:            cfg,
 		eventEnrichers: groupEventEnrichers(cfg.EventEnrichers),
 	}
 }
 
-func groupEventEnrichers(enrichers []EventEnricher) map[castpb.EventType][]EventEnricher {
-	result := map[castpb.EventType][]EventEnricher{}
-
+func groupEventEnrichers(enrichers []EventEnricher) map[events.ID][]EventEnricher {
+	result := map[events.ID][]EventEnricher{}
 	for _, enricher := range enrichers {
 		for _, t := range enricher.EventTypes() {
 			result[t] = append(result[t], enricher)
 		}
 	}
-
 	return result
 }
 
 type Service struct {
-	log         *logging.Logger
```

Comment:
Need to rollback this change and keep enrichers as is and refactor to support pb.ContainerEvent proto instead of pb.Event